packages/graphback-keycloak-authz/src/utils.ts

```diff
@@ -0,0 +1,36 @@
+import { ModelAuthRule, ModelOperation, CrudServiceAuthConfig } from "./definitions"
+
+
+const supportedCrudOperations = ['create', 'read', 'update', 'delete'];
+
+export function getOperationsFromRule(rule: ModelAuthRule): ModelOperation[] {
+  if (rule.operations && rule.operations.length > 0){
+    const invalidOperation = rule.operations.find((operation: string) => !supportedCrudOperations.includes(operation));
+    if (invalidOperation) {
+      throw new Error(`auth rule ${JSON.stringify(rule)} is invalid\n"${invalidOperation}" is not a supported operation`);
+    }
+
+    return rule.operations;
+  }
+
+  return supportedCrudOperations as ModelOperation[];
+}
+
+export function getEmptyServiceConfig() {
+  const serviceConfig = {} as CrudServiceAuthConfig;
+  for (const operation of supportedCrudOperations) {
+    serviceConfig[operation] = { roles: [] };
+  }
+
+  return serviceConfig;
+}
+
+export function isAuthorizedByRole(roles: string[], context?: any) {
+  for (const role of roles) {
+    if (context.kauth.hasRole(role)) {
```

Comment:
So this package will require our keycloak library as an indirect dependency? Is there easy win to document this/add keycloak-connect-graphql to dependencies?

This is about: How to connect that to the `context.kauth` with indirect dependency

Reply:
Yeah understood. I'll have a think about it. Perhaps this package could take the keycloak-connect-graphql lib as a dependency and rexport it and then we have tests to verify the integration. I will think of other ideas.

Reply:
No strong opinion on this - if you manage to document it right (as this will be added manually by users so they need some clear docs for this

Reply:
Yeah totally agreed!